caddyhttp/httpserver/plugin_test.go

```diff
@@ -80,16 +84,37 @@ func TestStandardizeAddress(t *testing.T) {
 	}
 }
 
+func TestHostIsIDN(t *testing.T) {
+	for i, test := range []struct {
+		host     string
+		expected bool
+	}{
+		{"idnhost-öåä", true},
+		{"host", false},
+		{"", false},
+		{"1.2.3.4", false},
+		{"::1", false},
+		{"[::1]", false},
+	} {
+		actual := isIDN(test.host)
+		if actual != test.expected {
+			t.Errorf("Test %d: expected '%t' but got '%t'", i, test.expected, actual)
+		}
+	}
+}
+
 func TestAddressVHost(t *testing.T) {
 	for i, test := range []struct {
 		addr     Address
 		expected string
 	}{
-		{Address{Original: "host:1234"}, "host:1234"},
-		{Address{Original: "host:1234/foo"}, "host:1234/foo"},
-		{Address{Original: "host/foo"}, "host/foo"},
-		{Address{Original: "http://host/foo"}, "host/foo"},
-		{Address{Original: "https://host/foo"}, "host/foo"},
+		{Address{Original: "host:1234", Scheme: "", Host: "host", Port: "1234", Path: ""}, "host:1234"},
+		{Address{Original: "host:1234/foo", Scheme: "", Host: "host", Port: "1234", Path: "/foo"}, "host:1234/foo"},
+		{Address{Original: "host/foo", Scheme: "", Host: "host", Port: "", Path: "/foo"}, "host/foo"},
+		{Address{Original: "http://host/foo", Scheme: "http", Host: "host", Port: "", Path: "/foo"}, "host/foo"},
+		{Address{Original: "https://host/foo", Scheme: "https", Host: "host", Port: "", Path: "/foo"}, "host/foo"},
```

Comment:
I am not sure rewriting these existing tests is desirable, or at least, doesn't seem relevant for this change.